grpc/src/client.rs

```diff
@@ -0,0 +1,642 @@
+use std::error::Error as StdError;
+use std::fmt;
+
+use ::tendermint::chain::Id;
+use bytes::Bytes;
+use celestia_types::any::IntoProtobufAny;
+use http_body::Body;
+use k256::ecdsa::VerifyingKey;
+use lumina_utils::time::Interval;
+use prost::Message;
+use signature::Keypair;
+use std::time::Duration;
+use tokio::sync::{MappedMutexGuard, Mutex, MutexGuard};
+use tonic::body::Body as TonicBody;
+use tonic::client::GrpcService;
+use tonic::codegen::Service;
+
+use celestia_grpc_macros::grpc_method;
+use celestia_proto::celestia::blob::v1::query_client::QueryClient as BlobQueryClient;
+use celestia_proto::celestia::core::v1::gas_estimation::gas_estimator_client::GasEstimatorClient;
+use celestia_proto::celestia::core::v1::tx::tx_client::TxClient as TxStatusClient;
+use celestia_proto::cosmos::auth::v1beta1::query_client::QueryClient as AuthQueryClient;
+use celestia_proto::cosmos::bank::v1beta1::query_client::QueryClient as BankQueryClient;
+use celestia_proto::cosmos::base::node::v1beta1::service_client::ServiceClient as ConfigServiceClient;
+use celestia_proto::cosmos::base::tendermint::v1beta1::service_client::ServiceClient as TendermintServiceClient;
+use celestia_proto::cosmos::staking::v1beta1::query_client::QueryClient as StakingQueryClient;
+use celestia_proto::cosmos::tx::v1beta1::service_client::ServiceClient as TxServiceClient;
+use celestia_types::blob::{BlobParams, MsgPayForBlobs, RawBlobTx, RawMsgPayForBlobs};
+use celestia_types::block::Block;
+use celestia_types::consts::appconsts;
+use celestia_types::hash::Hash;
+use celestia_types::state::auth::{Account, AuthParams, BaseAccount};
+use celestia_types::state::{
+    AbciQueryResponse, PageRequest, QueryDelegationResponse, QueryRedelegationsResponse,
+    QueryUnbondingDelegationResponse, RawTxBody, ValAddress,
+};
+use celestia_types::state::{
+    AccAddress, Address, AddressTrait, Coin, ErrorCode, TxResponse, BOND_DENOM,
+};
+use celestia_types::{AppVersion, Blob, ExtendedHeader};
+
+use crate::abci_proofs::ProofChain;
+use crate::grpc::{
+    BroadcastMode, GasEstimate, GasInfo, GetTxResponse, TxPriority, TxStatus, TxStatusResponse,
+};
+use crate::signer::{sign_tx, DispatchedDocSigner, KeypairExt};
+use crate::tx::TxInfo;
+use crate::{Error, Result, TxConfig};
+
+// source https://github.com/celestiaorg/celestia-core/blob/v1.43.0-tm-v0.34.35/pkg/consts/consts.go#L19
+const BLOB_TX_TYPE_ID: &str = "BLOB";
+// Multiplier used to adjust the gas limit given by gas estimation service
+const DEFAULT_GAS_MULTIPLIER: f64 = 1.1;
+
+struct AccountBits {
+    account: Account,
+    app_version: AppVersion,
+    chain_id: Id,
+}
+
+pub(crate) struct SignerBits {
+    pub(crate) signer: DispatchedDocSigner,
+    pub(crate) pubkey: VerifyingKey,
+}
```

Comment:
could we `s/Bits/Config/g` or `Conf`? I tried to get used to this name but my mind keeps connecting it to bytes somehow

Reply:
Settled on `SignerConfig` and `AccountState`, since the latter isn't really a config, but a cached account state we're using for txns.